### PR TITLE
[processor/transform] Fix trace accessor bugs

### PR DESCRIPTION
**Pull request by TylerHelmuth**
**Description:**
Some accessors for traces were not returned proper data types, breaking their usage in conditions.  This PR fixes those bugs.

**Link to tracking Issue:**
#10350 

**Testing:**
Updated/Added unit tests

### CODE REVIEW

File: processor/transformprocessor/internal/common/functions_test.go
Comment (codeboten) [inline processor/transformprocessor/internal/common/functions_test.go:214]: Please split off this change (strp -> testhelper.Strp) into its own PR to fix the broken tests

File: processor/transformprocessor/internal/traces/traces.go
Comment (bogdandrutu) [inline processor/transformprocessor/internal/traces/traces.go:218]: Instead of this (it allocates extra string) can we determine that it checks for "trace_id" and convert that to TraceID? Or maybe change the query to be `span_id == span_id("0102030405060708")`?

Comment (TylerHelmuth) [reply processor/transformprocessor/internal/traces/traces.go:218]: @bogdandrutu opened https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/10487 to add SpanIDs and TraceIDs to the grammar.  Once that PR is merged, we'd be able to have the accessor's Getter return the literal SpanID/TraceID and Setter accept a literal SpanID/TraceID instead of a string.

**Review by anuraaga — APPROVED** [review]

File: processor/transformprocessor/internal/traces/traces.go
Comment (codeboten) [reply processor/transformprocessor/internal/traces/traces.go:222]: Should this be updated now that #10487 merged?

Comment (TylerHelmuth) [reply processor/transformprocessor/internal/traces/traces.go:222]: Yes I haven't gotten back to the PR yet, got pulled away but some other work.  I plan to work on it today.

Comment (TylerHelmuth) [reply processor/transformprocessor/internal/traces/traces.go]: Done.  Traces now returns/accepts SpanID/TraceID in the accessors.

**Comment by TylerHelmuth** [comment]
@open-telemetry/collector-contrib-approvers please review

**Comment by TylerHelmuth** [comment]
https://github.com/orgs/open-telemetry/teams/collector-contrib-approvers please review

**Review by mx-psi — APPROVED** [review]

**Pull request merged**
